codec/src/main/java/io/netty/handler/codec/marshalling/MarshallingDecoder.java

```diff
@@ -80,8 +80,17 @@ protected Object decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception
         }
     }
 
-    @Override
-    protected ByteBuf extractFrame(ChannelHandlerContext ctx, ByteBuf buffer, int index, int length) {
+    protected ByteBuf extractFrame(ByteBuf buffer, int index, int length) {
         return buffer.slice(index, length);
     }
+
+    /**
+     * @deprecated - use {@link #extractFrame(ByteBuf, int, int)} instead, since the
+     * {@link ChannelHandlerContext} is not actually needed for implementation
+     */
+    @Deprecated
+    protected ByteBuf extractFrame(@SuppressWarnings("unused") ChannelHandlerContext ignored, ByteBuf buffer, int index,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1161)